onedal/neighbors/neighbors.py

```diff
@@ -582,6 +582,8 @@ def _onedal_predict(self, model, X, params, queue):
             model = self._onedal_model
         else:
             model = self._create_model(backend)
+        if 'responses' not in params['result_option'] and gpu_device:
+            params['result_option'] += '|responses'
```

Comment:
Please move kNN fix into separate PRs. I think it requires separate review.
[Generally, this part of code should be refactored with no explicitly getting sycl queue device here. This device support logic with fallbacks to daal4py should be resolved on `gpu_support` or `cpu_support` methods like we have for other algos for consistency.] So it is better to move this discussion into separate PR

Reply:
Done - https://github.com/intel/scikit-learn-intelex/pull/1337. I think both are ready for approval/merge.